File: app/src/main/java/zone/com/zrefreshlayoutdemo/MainActivity.java
package zone.com.zrefreshlayoutdemo;

import android.content.Intent;
import android.graphics.Color;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.View;
import android.widget.TextView;

import butterknife.ButterKnife;
import butterknife.OnClick;
import zone.com.zrefreshlayoutdemo.common.Constant;
import zone.com.zrefreshlayoutdemo.common.HeadSetting;

public class MainActivity extends AppCompatActivity {


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.main);
        ButterKnife.bind(this);

    }

    @OnClick({R.id.textView, R.id.listView, R.id.recyclerView, R.id.tv_mode, R.id.globalView,
            R.id.gridView, R.id.scrollerView, R.id.webView,
            R.id.autoRefresh, R.id.pinContent, R.id.config,
            R.id.uniqueFeature, R.id.refreshAblePosition, R.id.resistance,
            R.id.waveHeader, R.id.circleHeader,
            R.id.sinaRefresh, R.id.meterialHeader,
            R.id.cirlcleActivity, R.id.nestViewActivity,
            R.id.nestScroller,
            R.id.flexibilityListViewActivity, R.id.nestedScrollingActivity_Parent
    })
    public void onClick(View view) {
        if(view.getId()== R.id.globalView)
            Apps.setGlobalHead(SP1.INSTANCE.get(Constant.REFRESH_MODE,HeadSetting.class));
        else
            Apps.setDefaultHeader();

        switch (view.getId()) {
            case R.id.globalView:
                startActivity(new Intent(this, GlobalViewActivity.class));
                break;
            case R.id.tv_mode:
                startActivity(new Intent(this, SettingActivity.class));
                break;
            case R.id.textView:
                startActivity(new Intent(this, TextViewActivity.class));
                break;
            case R.id.listView:
                startActivity(new Intent(this, ListViewActivity.class));
                break;
            case R.id.recyclerView:
                startActivity(new Intent(this, RecyclerViewActivity.class));
                break;
            case R.id.gridView:
                startActivity(new Intent(this, GridViewActivity.class));
                break;
            case R.id.scrollerView:
                startActivity(new Intent(this, ScrollerViewActivity.class));
                break;
            case R.id.webView:
                startActivity(new Intent(this, WebViewActivity.class));
                break;
            case R.id.autoRefresh:
                startActivity(new Intent(this, AutoRefreshActivity.class));
                break;
            case R.id.pinContent:
                startActivity(new Intent(this, PinContentActivity.class));
                break;
            case R.id.uniqueFeature:
                startActivity(new Intent(this, UniqueFeatureActivity.class));
                break;
            case R.id.refreshAblePosition:
                startActivity(new Intent(this, RefreshPositionActivity.class));
                break;
            case R.id.resistance:
                startActivity(new Intent(this, ResistanceActivity.class));
                break;
            case R.id.waveHeader:
                startActivity(new Intent(this, WaveHeaderActivity.class));
                break;
            case R.id.circleHeader:
                startActivity(new Intent(this, CircleHeaderActivity.class));
                break;
            case R.id.sinaRefresh:
                startActivity(new Intent(this, SinaHeaderActivity.class));
                break;
            case R.id.meterialHeader:
                startActivity(new Intent(this, MeterialHeaderActivity.class));
                break;
            case R.id.cirlcleActivity:
                startActivity(new Intent(this, CirlcleActivity.class));
                break;
            case R.id.nestViewActivity:
                startActivity(new Intent(this, NestViewActivity.class));
                break;
            case R.id.flexibilityListViewActivity:
                startActivity(new Intent(this, FlexibilityListViewActivity.class));
                break;
            case R.id.nestedScrollingActivity_Parent:
                startActivity(new Intent(this, NestedScrollingActivity_Parent.class));
                break;
            case R.id.nestScroller:
                startActivity(new Intent(this, ScrollerNesttActivity.class));
                break;
        }
    }

}

File: app/src/main/java/zone/com/zrefreshlayoutdemo/MainActivity.kt
package zone.com.zrefreshlayoutdemo

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import android.view.View
import com.zone.lib.LogZSDK
import com.zone.lib.ZLogger
import com.zone.lib.base.controller.activity.BaseFeatureActivity
import com.zone.lib.base.controller.activity.controller.SwipeBackActivityController
import zone.com.zrefreshlayoutdemo.common.Constant
import zone.com.zrefreshlayoutdemo.common.HeadSetting

class MainActivity : BaseFeatureActivity() {
    companion object {
        //还原最开始的log配置  如果某次配置一次特殊的 打印完后的记得还原配置
        @JvmStatic
        fun initLogger() {
            ZLogger.logLevelList.clear()
            ZLogger.mayLoggerList.clear()
//            ZLogger.mayLoggerList.addAll(listOf<ZLogger>(LogCustomView, LogZSDK))
        }
    }

    override fun initDefaultConifg() {
        unRegisterPrestener(SwipeBackActivityController::class.java)
    }

    override fun initData() {
        permissionCheckStorageMustPermit()
        setContentView(R.layout.main)
    }
    override fun onStoragePermit() {
        super.onStoragePermit()
    }

    override fun onStorageDeniedMustPermit() {
        super.onStorageDeniedMustPermit()
        finish()
    }


    override fun setContentView() {
    }

    override fun setListener() {
    }


    override fun onClick(view: View?) {
        view?.let {
            if (it.id == R.id.globalView)
                Apps.setGlobalHead(SP1.get(Constant.REFRESH_MODE, HeadSetting::class.java))
            else
                Apps.setDefaultHeader()

            when (it.id) {
                R.id.globalView -> startActivity(Intent(this, GlobalViewActivity::class.java))
                R.id.tv_mode -> startActivity(Intent(this, SettingActivity::class.java))
                R.id.textView -> startActivity(Intent(this, TextViewActivity::class.java))
                R.id.listView -> startActivity(Intent(this, ListViewActivity::class.java))
                R.id.recyclerView -> startActivity(Intent(this, RecyclerViewActivity::class.java))
                R.id.gridView -> startActivity(Intent(this, GridViewActivity::class.java))
                R.id.scrollerView -> startActivity(Intent(this, ScrollerViewActivity::class.java))
                R.id.webView -> startActivity(Intent(this, WebViewActivity::class.java))
                R.id.autoRefresh -> startActivity(Intent(this, AutoRefreshActivity::class.java))
                R.id.pinContent -> startActivity(Intent(this, PinContentActivity::class.java))
                R.id.uniqueFeature -> startActivity(Intent(this, UniqueFeatureActivity::class.java))
                R.id.refreshAblePosition -> startActivity(Intent(this, RefreshPositionActivity::class.java))
                R.id.resistance -> startActivity(Intent(this, ResistanceActivity::class.java))
                R.id.waveHeader -> startActivity(Intent(this, WaveHeaderActivity::class.java))
                R.id.circleHeader -> startActivity(Intent(this, CircleHeaderActivity::class.java))
                R.id.sinaRefresh -> startActivity(Intent(this, SinaHeaderActivity::class.java))
                R.id.meterialHeader -> startActivity(Intent(this, MeterialHeaderActivity::class.java))
                R.id.cirlcleActivity -> startActivity(Intent(this, CirlcleActivity::class.java))
                R.id.flexibilityListViewActivity -> startActivity(Intent(this, FlexibilityKtActivity::class.java))
                R.id.nestScroller -> startActivity(Intent(this, ScrollerNesttActivity::class.java))
                else -> {
                }
            }
        }

    }

}
